### PR TITLE
Feature/server/e3 expense tracking form

### PR DESCRIPTION
**Pull request by Atsuki02**
## Describe your changes

- Create delete-expense-api
- Create create-expense-api
- Create edit-expense-api
- Create update-payments-per-date api
- Create update-payments-per-month api

## Issue

#95 
#97 
#99 
#101 

## note

I'll make get Expense api in another PR

## URL
none

## Screenshot (if you made the UI)
none

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings and no errors

### CODE REVIEW

File: server/src/prisma/seedData/expense.ts [hunk 1]
```diff
@@ -35,8 +35,8 @@ export const expenseData = async () => [
     id: 115,
     itemName: 'Maintenance Fee',
     fee: 150.0,
-    date: new Date(2023, 4, 30),
-    houseId: 110,
-    buyerId: 105,
+    date: new Date(2023, 3, 30),
+    houseId: 109,
+    buyerId: 104,
   },
```
Comment (Atsuki02) [inline server/src/prisma/seedData/expense.ts:41]: I changed seed data to test easily

File: server/src/expense/expense.service.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  UnauthorizedException,
+} from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { CreateExpenseDto, UpdateExpenseDto } from './dto';
+
+@Injectable()
+export class ExpenseService {
+  constructor(private prisma: PrismaService) {}
+
+  async createExpense(userId: number, dto: CreateExpenseDto) {
```
Comment (kanatagu) [inline server/src/expense/expense.service.ts:13]: You need to validate if the user belongs to the house using userId and houseId!

File: server/src/expense/expense.service.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+          expense: { ...createdExpense, payments: createdPaymentsResult },
+        };
+      });
+    } catch (error) {
+      console.error('Error creating expense:', error);
+      throw error;
+    }
+  }
+
+  async updateExpense(expenseId: number, dto: UpdateExpenseDto) {
```
Comment (kanatagu) [inline server/src/expense/expense.service.ts:66]: You also need validation that only the user who created the expense can update!

File: server/src/payment/payment.service.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import {
+  UpdatePaymentsStatusPerDateDto,
+  UpdatePaymentsStatusPerMonthDto,
+} from './dto';
+
+@Injectable()
+export class PaymentService {
+  constructor(private prisma: PrismaService) {}
+
+  async updatePaymentsStatusPerDate(dto: UpdatePaymentsStatusPerDateDto) {
```
Comment (kanatagu) [inline server/src/payment/payment.service.ts:12]: Add validation for the user belongs to the house same as perMonth!

File: server/src/expense/expense.service.ts (hunk 2 above)
Comment (Atsuki02) [reply server/src/expense/expense.service.ts:14]: Fixed!
https://github.com/HAK2024/fairy-share/pull/102/commits/1f872e37dae2b6b822a67d341b0f6e8a8ba6e451

File: server/src/expense/expense.service.ts (hunk 3 above)
Comment (Atsuki02) [reply server/src/expense/expense.service.ts]: Fixed it!
https://github.com/HAK2024/fairy-share/pull/102/commits/60b7dbcacb815f99cf11087e133978bd07de78fa

File: server/src/payment/payment.service.ts (hunk 4 above)
Comment (Atsuki02) [reply server/src/payment/payment.service.ts]: Fixed!
https://github.com/HAK2024/fairy-share/pull/102/commits/a1fc71f919bcd609119883c6483d1e4432799dc3

**Review by kanatagu — APPROVED** [review]
LGTM!

**Pull request merged**